### PR TITLE
Fix failing rspec tests

### PR DESCRIPTION
**Pull request by salimane**
cc @raphink

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -11,7 +11,6 @@
     @old_env = {}
     ENV.each_key {|k| @old_env[k] = ENV[k]}
 
-    c.strict_variables = Gem::Version.new(Puppet.version) >= Gem::Version.new('3.5')
```
Comment (raphink) [inline spec/spec_helper.rb]: This one should be kept

File: Gemfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 source ENV['GEM_SOURCE'] || "https://rubygems.org"
 
 group :development, :unit_tests do
-  gem 'rake',                                              :require => false
+  gem 'rake', '< 11.0.1',                                  :require => false
```
Comment (raphink) [inline Gemfile]: OK. I'll actually update that as we use modulesync and this change is to be applied.

File: spec/spec_helper.rb (hunk 1 above)
Comment (salimane) [inline spec/spec_helper.rb]: @raphink any reason for keeping this ?
Thanks

Comment (raphink) [inline spec/spec_helper.rb]: Yes, we want to test with strict variables whenever possible. Using strict variables avoids many a bug.

**Comment by raphink** [comment]
All these points are actually fixed already.

**Pull request closed without merging**